ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySuspendTest.java

```diff
@@ -0,0 +1,171 @@
+            })
+    public void testSuspensionWhenThereArentTransactionsToRecover() {
+
+        // Make sure that the test environment is ready
+        BytemanControlledRecord.resetCommitCallCounter();
+        // The following instruction makes the BytemanControlledRecord commit without failures.
+        BytemanControlledRecord.setGreenFlag();
+
+        createAtomicAction(new BytemanControlledRecord(true));
+
+        // The Transaction System needs to be disabled, i.e. no new transactions can be created
+        TxControl.disable();
+
+        // Makes sure that the transaction reaper completed all transactions
+        TransactionReaper.transactionReaper().waitForAllTxnsToTerminate();
+
+        // Synchronization is not needed (i.e. async = true) as isWaitForFinalRecovery() == true
+        _manager.suspend(true);
+
+        // BytemanControlledRecord.getCommitCallCounter() should be 1 as:
+        // - One invocation from the normal commit procedure should pass
+        Assertions.assertEquals(1, BytemanControlledRecord.getCommitCallCounter(),
+                String.format("BytemanControlledRecord's getCommitCallCounter is %d but it should have been 1",
+                        BytemanControlledRecord.getCommitCallCounter()));
+    }
+
+    @Test
+    @BMScript("recoverySuspendTest_BytemanControlledRecord")
+    public void testSuspensionWhenThereIsAtomicActionToRecover() {
+
+        // Make sure that the test environment is ready
+        BytemanControlledRecord.resetCommitCallCounter();
+        BytemanControlledRecord.resetGreenFlag();
+        
+        createAtomicAction(new BytemanControlledRecord(true));
+
+        // The Transaction System needs to be disabled, i.e. no new transactions can be created
+        TxControl.disable();
+
+        // Makes sure that the transaction reaper completed all transactions
+        TransactionReaper.transactionReaper().waitForAllTxnsToTerminate();
+
+        // Synchronization is not needed (i.e. async = true) as isWaitForFinalRecovery() == true
+        _manager.suspend(true);
+
+        // BytemanControlledRecord.getCommitCallCounter() should be 3 as:
+        // - One invocation from the normal commit procedure should fail
+        // - One invocation from the recovery process should fail
+        // - One invocation from the recovery process should pass
+        Assertions.assertEquals(3, BytemanControlledRecord.getCommitCallCounter(), 
+                String.format("BytemanControlledRecord's getCommitCallCounter is %d but it should have been 3",
+                        BytemanControlledRecord.getCommitCallCounter()));
+    }
+
+    private void createAtomicAction(AbstractRecord secondRecord) {
+
+        AtomicAction A = new AtomicAction();
+
+        BasicRecord basicRecordOne = new BasicRecord();
+
+        A.begin();
+
+        A.add(basicRecordOne);
+        A.add(secondRecord);
+
+        // should generate a recovery record
```

Comment:
Maybe change to: `// should generate a recovery record if NOT flagged("greenFlag") by the byteman rule` ?

I suppose it's correct in someways in both ways but I think this comment is more for the "crash" situation (which doesn't happen for both of the tests, just one of them)

Reply:
Yeah, it makes sense. If you agree, I will remove the comment entirely as it is misleading

Reply:
(I've already removed it but I can always add it back if you think it can help clarify the logic of that method)

Reply:
Thank you, this one can be marked resolved!